setup.py

```diff
@@ -31,14 +36,15 @@
             'backports.tempfile==1.0rc1',
             'cloudpickle>=0.1.0',
             'futures>=3.0.1',
-            'pylint',
+            'pylint~=2.7',
```

Comment:
please revert

Reply:
I added restrictive version requirements on all dependencies. I once again had several tests fail due to a change in the behavior of multiple dependencies (both isort and pylint). The tests did not behave the same on my laptop with an older version of isort/pylint and in the CI where the latest versions were downloaded.

Unless there are specific tests for multiple versions of a dependency, I don't think we should assume we support it.